pkg/reconciler/channel/channel.go

```diff
@@ -123,30 +124,42 @@ func (r *Reconciler) reconcile(ctx context.Context, c *v1alpha1.Channel) error {
 		return nil
 	}
 
-	channelResourceInterface := r.DynamicClientSet.Resource(duckroot.KindToResource(c.Spec.ChannelTemplate.GetObjectKind().GroupVersionKind())).Namespace(c.Namespace)
+	gvr, _ := meta.UnsafeGuessKindToResource(c.Spec.ChannelTemplate.GetObjectKind().GroupVersionKind())
+	channelResourceInterface := r.DynamicClientSet.Resource(gvr).Namespace(c.Namespace)
+	if channelResourceInterface == nil {
+		msg := fmt.Sprintf("Unable to create dynamic client for: %+v", c.Spec.ChannelTemplate)
+		logging.FromContext(ctx).Error(msg)
```

Comment:
We don't need this, the outer `Reconcile` method will log the error.

Reply:
Removed, thanks!